backend/src/services/episode/episode.service.ts

```diff
@@ -245,6 +245,13 @@ class Episode {
 
     return episode;
   }
+
+  public async deleteAllByPodcastId(id: number): Promise<void> {
+    const episodes = await this.#episodeRepository.getAllByPodcastId(id);
+    episodes.forEach((episode) => {
+      this.delete(episode.id);
+    });
```

Comment:
```suggestion
    for (let i = 0; i < episodes.length; i++) {
      await this.delete(episodes[i].id);
    }
```
We should await before all episodes will be deleted, and only then move on

Reply:
Good catch, but we can do it better
```tsx
await Promise.all(episodes.map((it) => this.delete(it.id)))
```